src/sentry/tasks/sentry_apps.py

```diff
@@ -303,22 +303,25 @@ def send_and_save_webhook_request(url, sentry_app, app_platform_event):
         resp = safe_urlopen(
             url=url, data=app_platform_event.body, headers=app_platform_event.headers, timeout=5
         )
+        track_response_code(resp.status_code, slug, event)
+        buffer.add_request(
+            response_code=resp.status_code,
+            org_id=org_id,
+            event=event,
+            url=url,
+            error_id=resp.headers.get("Sentry-Hook-Error"),
+            project_id=resp.headers.get("Sentry-Hook-Project"),
+        )
+        resp.raise_for_status()
+        return resp
 
-    except RequestException:
+    except Timeout:
         track_response_code("timeout", slug, event)
         # Response code of 0 represents timeout
         buffer.add_request(response_code=0, org_id=org_id, event=event, url=url)
         # Re-raise the exception because some of these tasks might retry on the exception
         raise
-    track_response_code(resp.status_code, slug, event)
-
-    buffer.add_request(
-        response_code=resp.status_code,
-        org_id=org_id,
-        event=event,
-        url=url,
-        error_id=resp.headers.get("Sentry-Hook-Error"),
-        project_id=resp.headers.get("Sentry-Hook-Project"),
-    )
 
-    return resp
+    except HTTPError:
```

Comment:
@NisanthanNanthakumar There's no reason to catch the error here and do nothing except re-raise it again

Reply:
@scefali Makes sense. But should we try to catch any exceptions that may be caused by datadog or the logger. Maybe instead of the HTTPError, use the general Exception and re-raise?

Reply:
@NisanthanNanthakumar I don't think there is a reason to catch any error from datadog or the logger. We didn't have anything before. Sometimes it's good to re-raise a different error, but that's only if the calling function is expecting specific error (like an `IntegrationError`). Don't think that's happening here so we should just let the caller deal with whatever error was originally raised.

Reply:
@scefali Ok sounds good